### PR TITLE
Add Datasets Card spec

### PR DESCRIPTION
**Pull request by lhoestq**
I added a datasetcard spec in the same format as the one for models at https://github.com/huggingface/hub-docs/blob/main/modelcard.md

This can also be helpful for https://github.com/huggingface/huggingface_hub/pull/1282 cc @julien-c 
If this gets merged I'll also link the `datasets` docs to it

### CODE REVIEW

File: datasetcard.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+# Full dataset card template at https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/datasetcard_template.md
+language:
+- {lang_0}  # Example: fr
+- {lang_1}  # Example: en
+license: {license}  # Example: apache-2.0 or any license from https://hf.co/docs/hub/repositories-licenses
+tags:
+- {tag_0}  # Example: audio
+- {tag_1}  # Example: bio
+- {tag_2}  # Example: natural language understanding
+- {tag_3}  # Example: birds classification
```
Comment (julien-c) [inline datasetcard.md:12]:
```suggestion
- {tag_2}  # Example: natural-language-understanding
- {tag_3}  # Example: birds-classification
```

we don't have a strong convention but usually we try to have dash separated ids

File: datasetcard.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+annotations_creators:
+- {creator_0}  # Example: crowdsourced
+- {creator_1}  # Example: found
+- {creator_3}  # Example: expert-generated
+- {creator_4}  # Example: machine-generated
+language_creators:
+- {creator_0}  # Example: crowdsourced
+- {creator_1}  # Example: found
+- {creator_3}  # Example: expert-generated
+- {creator_4}  # Example: machine-generated
```
Comment (julien-c) [inline datasetcard.md:22]: most of those we would have only one value, so i would showcase just one elem:
```suggestion
annotations_creators:
- {creator}  # Example: crowdsourced, found, expert-generated, machine-generated
language_creators:
- {creator}  # Example: crowdsourced, ...
```

File: datasetcard.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+paperswithcode_id: {paperswithcode_id}  # Dataset id on PapersWithCode (from the URL). Example for SQuAD: squad
+configs:  # Optional for datasets with multiple configurations like glue.
+- {config_0}  # Example for glue: sst2
+- {config_1}  # Example for glue: cola
+
+# Optional. This part can be used to store the feature types and size of the dataset to be used in python. This can be automatically generated using the datasets-cli.
+dataset_info:
+  features:
+    - name: {feature_name_0}    # Example: id
+      dtype: {feature_dtyp_e0}  # Example: int32
```
Comment (julien-c) [inline datasetcard.md:49]:
```suggestion
      dtype: {feature_dtype_0}  # Example: int32
```

File: datasetcard.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    #       dtype: string
+    #     - name: answer_start
+    #       dtype: int32
+  config_name: {config_name}  # Example for glue: sst2
+  splits:
+    - name: {split_name_0}                  # Example: train
+      num_bytes: {split_num_bytes_0}        # Example for SQuAD: 79317110
+      num_examples: {split_num_examples_0}  # Example for SQuAD: 87599
+  download_size: {dataset_download_size}   # Example for SQuAD: 35142551
+  dataset_size: {dataset_size}             # Example for SQuAD: 89789763
```
Comment (julien-c) [inline datasetcard.md:75]: mention that this will be auto-computed soon? so maybe no need to add it?

Comment (julien-c) [reply datasetcard.md:75]: oh actually there's a comment above that says this is generated automatically. so i think it's fine

**Review by julien-c — COMMENTED** [review]
I'll like this a lot!

I think we discussed this somewhere, didn't we? Do you have a Slack or GitHb link handy maybe?

File: datasetcard.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+# dataset_info:
+#   - config_name: {config0}
+#     features:
+#       ...
+#   - config_name: {config1}
+#     features:
+#       ...
+# ```
+
+# Optional. If you want your dataset to be protected behind a gate that users have to accept to access the dataset:
```
Comment (osanseviero) [inline datasetcard.md:88]: For this maybe link to https://huggingface.co/docs/hub/datasets-gated

File: datasetcard.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+extra_gated_fields:
+- {field_name_0}: {field_type_0}  # Example: Name: text
+- {field_name_1}: {field_type_1}  # Example: Affiliation: text
+- {field_name_2}: {field_type_2}  # Example: Email: text
+- {field_name_3}: {field_type_3}  # Example for speech datasets: I agree to not attempt to determine the identity of speakers in this dataset: checkbox
+extra_gated_prompt: {extra_gated_prompt}  # Example for speech datasets: By clicking on “Access repository” below, you also agree to not attempt to determine the identity of speakers in the dataset.
+
+# Optional. Add this if you want to encode a train and evaluation info in a structured way for AutoTrain
```
Comment (osanseviero) [inline datasetcard.md:96]: Any doc you can link for this?

**Review by osanseviero — COMMENTED** [review]
Very nice! :fire:

File: datasetcard.md (hunk 6 above)
Comment (lhoestq) [reply datasetcard.md:90]: I haven't found any doc about `train-eval-index`. Maybe @lewtun @abhishekkrthakur know more

File: datasetcard.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+license_details: {license_details}  # Example: content or link to a license not present in https://hf.co/docs/hub/repositories-licenses
+pretty_name: {pretty_name}  # Example: SQuAD
+size_categories:
+- {size_0}  # Example: n<1K
+- {size_0}  # Example: 100K<n<1M
+source_datasets:
+- {source_dataset_0}  # Example: wikipedia
+- {source_dataset_1}  # Example: laion/laion-2b
+task_categories:  # Full list at https://github.com/huggingface/hub-docs/blob/main/js/src/lib/interfaces/Types.ts
+- {task_0}  # Example: question-ansering
```
Comment (julien-c) [inline datasetcard.md:29]:
```suggestion
- {task_0}  # Example: question-answering
```

File: datasetcard.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+language_creators:
+- {creator}  # Example: crowdsourced, ...
+language_details:
+- {bcp47_lang_0}  # Example: fr-FR
+- {bcp47_lang_1}  # Example: en-US
+license_details: {license_details}  # Example: content or link to a license not present in https://hf.co/docs/hub/repositories-licenses
+pretty_name: {pretty_name}  # Example: SQuAD
+size_categories:
+- {size_0}  # Example: n<1K
+- {size_0}  # Example: 100K<n<1M
```
Comment (julien-c) [inline datasetcard.md:24]:
```suggestion
- {number_of_elements_in_dataset}  # Example: n<1K, 100K<n<1M, …
```

**Review by julien-c — APPROVED** [review]
with a few final suggestions, LGTM

File: datasetcard.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+- {tag_2}  # Example: natural-language-understanding
+- {tag_3}  # Example: birds-classification
+annotations_creators:
+- {creator}  # Example: crowdsourced, found, expert-generated, machine-generated
+language_creators:
+- {creator}  # Example: crowdsourced, ...
+language_details:
+- {bcp47_lang_0}  # Example: fr-FR
+- {bcp47_lang_1}  # Example: en-US
+license_details: {license_details}  # Example: content or link to a license not present in https://hf.co/docs/hub/repositories-licenses
```
Comment (julien-c) [inline datasetcard.md:20]: place this right after `license:` no?

**Pull request merged**